libglusterfs/src/common-utils.c

```diff
@@ -1179,18 +1179,17 @@ _gf_string2long(const char *str, long *n, int base)
     old_errno = errno;
     errno = 0;
     value = strtol(str, &tail, base);
-    if (str == tail)
+    if ((str == tail) || (*tail != 0)) {
         errno = EINVAL;
```

Comment:
Since overflow is also caused by an invalid input, can we make this condition:

if ((str == tail) || (*tail != 0) || errno == ERANGE) and get rid of the redundant if (errno == ERANGE || errno == EINVAL) (line 1187-1188)?